### DISCUSSION TITLE
Hierarchical models via multiple GGraph instances?

### DISCUSSION THREAD

**Discussion by LiamWalsh98** [eclipse-glsp/glsp#446] [Q&A] [answered]
Hello! I came across a discussion post (#353) while trying to research the same use case it proposes. I was wondering if there is a publicly accessible example somewhere using the "workaround number 1" in the provided answer (Navigating between separate diagrams, presumably GGraph objects?). 

This solution sounds perfect for what I am attempting to implement with GLSP (that being UML-RT). Though, the concept seems outside the scope of current examples that I've come across in my attempted adaptation.

From what I can already tell, I'll probably need to represent certain objects as both GNodes and GGraphs, in order to have this work. I'm also not certain what would be the best way to cache these objects. Finally, I'm not a very experienced front-end developer, so the client plugin would probably be the hardest part. Is there an existing GLSP message/call that can ask the server to switch the graph view within the same model? (Perhaps as part of the context menu?).

Any info on this particular implementation/workaround would be greatly appreciated!

**Comment by planger** [comment, marked as answer]
Hi,

I'm afraid there is no publicly accessible example. However, I would recommend to proceed as follows:
- Define a trigger when you want to switch into the other diagram, e.g. a [context menu entry](https://github.com/eclipse-glsp/glsp-server/blob/742be5d6f124aa9b84c480d40554a0f8f243fcb9/plugins/org.eclipse.glsp.server/src/org/eclipse/glsp/server/di/DiagramModule.java#L265) on the parent element, or a double-click listener, or reacting to a zoom event.
- Send a `RequestModelAction` from the client to the server on the aforementioned event. You can use the `RequestModelAction.options` to specify the context (e.g. the ID of the element for which you want to request the detail model).
- On the server, you can then react to the `RequestModelAction` e.g. in your [ModelSourceLoader](https://github.com/eclipse-glsp/glsp-server/blob/master/plugins/org.eclipse.glsp.server/src/org/eclipse/glsp/server/features/core/model/ModelSourceLoader.java) and construct the detail model (GModel) and send it to the client via the `ModelSubmissionHandler`.

This will result in a new rendering of the detail diagram whenever the trigger defined in the first step is activated.
There are for sure a few details to figure out along the way, but overall this should do the trick.

**Reply by LiamWalsh98** [reply]
Thanks very much for the response! I'll look into implementing it this way.
